libs/vue/src/schematics/component/schematic.ts

```diff
@@ -0,0 +1,90 @@
+import { strings } from '@angular-devkit/core';
+import {
+  apply,
+  applyTemplates,
+  chain,
+  mergeWith,
+  move,
+  Rule,
+  SchematicContext,
+  Tree,
+  url,
+} from '@angular-devkit/schematics';
+import {
+  formatFiles,
+  getProjectConfig,
+  offsetFromRoot,
+  projectRootDir,
+  ProjectType,
+  toClassName,
+  toFileName,
+} from '@nrwl/workspace';
+import { ComponentSchematicSchema } from './schema';
+
+interface NormalizedSchema extends ComponentSchematicSchema {
+  componentName: string;
+  projectRoot: string;
+  componentPath: string;
+  styleExtension: string;
+  destinationPath: string;
+}
+
+function normalizeOptions(
+  host,
+  options: ComponentSchematicSchema,
+  context: SchematicContext
+): NormalizedSchema {
+  const componentName = toClassName(options.name);
+
+  const componentPath = options.directory
+    ? `${toFileName(options.directory)}`
+    : '';
+
+  const projectName = options.project;
+  const projectConfig = getProjectConfig(host, projectName);
+
+  const projectRoot = `${projectRootDir(projectConfig.projectType)}/${
+    options.project
+  }`;
+
+  // depending on projectType build destination path of component
+  const destinationPath =
+    projectConfig.projectType === ProjectType.Application
+      ? `${projectRoot}/src/components/${componentPath}`
```

Comment:
I think `components/` should be removed because users may want to put components in `containers/`, `pages/`, or some other directory.

Reply:
I created a vue app first and inside that schematic the HelloWorld Component is inside a components folder